src/Api/Response/Parser/FiltersParser.php

```diff
@@ -70,7 +73,28 @@ public function parseForWidgets(SimpleXMLElement $data): array
             return [];
         }
 
-        // Your implementation.
+        foreach ($filters as &$filter) {
+            if (!isset($filter['values']) || !$filter['values']) {
+                continue;
+            }
+
+            switch ($filter['id']) {
+                case 'vendor':
+                    $filter['type'] = 'producer';
+                    break;
+                case 'cat':
+                    $filter['type'] = 'category';
+                    break;
+                case 'price':
+                    $filter['type'] = 'price';
+                    break;
+                default:
+                    $filter['type'] = 'dynamic';
+                    break;
+            }
+        }
+
+        unset($filter);
```

Comment:
PHP should automatically garbage collect this, no need of manually unsetting that.

Reply:
Not for garbage collection, just a good practice when working with foreach loops and references. https://www.php.net/manual/en/control-structures.foreach.php "Warning Reference of a $value and the last array element remain even after the foreach loop. It is recommended to destroy it by unset(). "

Reply:
Ok, since I really do not like unsetting such things manually, it may make sense to move the whole foreach into a separate method and create an array, without using references. Build the array yourself and return it then. I thought of something like that:

```php
function callThisSomethingReasonable(array $filters)
{
    $result = [];
    foreach ($filters as $filter) {
        // switch { ...
        // case: ...
            // $filter['manipulated'] = 'manipulated';
            $result[] = $filter;
    }
}
```

This is cleaner and prevents such things from even happening in the first place.

Reply:
I left it in the same function. A separate one seemed unnecessary as this is almost all that parseForWidgets does anyway.

Reply:
That's also fine for me.